internal/cli/push.go

```diff
@@ -10,7 +11,44 @@ import (
 
 var ()
 
-// CmdPush implements the epinio orgs command
+func init() {
+	CmdPush.Flags().StringSliceP("bind", "b", []string{}, "services to bind immediately")
+	CmdPush.RegisterFlagCompletionFunc("bind", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		// `cmd`, `args` are ignored.
+		// `toComplete` is the option value entered so far.
+		// This is a StringSlice option.
+		// This means that the option value is a comma-separated string of values.
+		// Completion has to happen only for the last segment in that string, i.e. after the last comma.
+		// Note that cobra does not feed us a slice, just the string.
+		// We are responsible for splitting into segments, and expanding only the last segment.
+
+		app, err := clients.NewEpinioClient(cmd.Flags())
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+
+		values := strings.Split(toComplete, ",")
+		if len(values) == 0 {
+			// Nothing. Report all possible matches
+			matches := app.ServiceMatching(toComplete)
+			return matches, cobra.ShellCompDirectiveNoFileComp
+		}
+
+		// Expand the last segment. The returned matches are
+		// the string with its last segment replaced by the
+		// expansions for that segment.
+
+		matches := []string{}
+		for _, match := range app.ServiceMatching(values[len(values)-1]) {
+			values[len(values)-1] = match
```

Comment:
It changes the list of (for example `['fox','d']` to `['fox','dog']` when `dog` is valid expansion/match for `d`.
The join then generates `fox,dog` from it, which goes into the slice of matches.
This is done for all expansions (dog, delphin, ...) of `d`...

Now that I know a bit more about `append` it could be done without overwriting, I guess.

Reply:
Leaving as is for now.

Reply:
gotcha, looks good.